unit_tests/sources/declarative/parsers/test_model_to_component_factory.py

```diff
@@ -2902,7 +2902,7 @@ def test_use_request_options_provider_for_datetime_based_cursor():
         parameters={},
     )
 
-    connector_builder_factory = ModelToComponentFactory(emit_connector_builder_messages=True)
```

Comment:
I could have updated the assertion to check for a `assert isinstance(retriever.stream_slicer, TestReadStreamSlicerDecorated)` or disable `emit_connector_builder_messages`. I arbitrarily chose the latter